File: Fuzzification Tool/Datasets/DatasetChooser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Datasets
{
    public enum DatasetType
    {
        Heart, Iris, Seeds, Skin, Wine, Yeast, Bupa, Balance, Vowel, Pima, Venicle, Wpbc, Wdbc
    }

    public static class DatasetChooser
    {
        public static DataSets ReadDatasetFromFile(DatasetType type, int datasetSize, int attributes, int inputAttributes, int outputAttributes, int outputIntervals, string filename)
        {
            DataSets dataset = null;
            switch (type)
            {
                case DatasetType.Iris:
                    dataset = (DataSets) new DatasetIris( datasetSize, attributes,  inputAttributes,  outputAttributes,  outputIntervals,  filename);
                    break;
                default:
                    return null;
            }

            dataset.InitializeDataset();
            dataset.NormalizeDataset();
            dataset.WriteInfoToFile();
            return dataset;
        }
        public static DataSets ReadDatasetFromFile(DatasetType type, string filename)
        {
            DataSets dataset = null;
            switch (type)
            {
                case DatasetType.Iris:
                    dataset = (DataSets) new DatasetIris( filename);
                    break;
                default:
                    return null;
            }

            dataset.InitializeDataset();
            dataset.NormalizeDataset();
            dataset.WriteInfoToFile();
            return dataset;
        }
    }

}

File: zdrojove kody v c++/DatasetChooser.cpp
#include "DatasetChooser.h"
#include "DataSets.h"
#include "DatasetIris.h"

Datasets::DataSets* Datasets::DatasetChooser::ReadDatasetFromFile(DatasetType type, int datasetSize, int inputAttributes, int outputIntervals, const std::wstring& filename)
	{
		DataSets* dataset = nullptr;
		switch (type)
		{
		case Datasets::DatasetType::Iris:
			dataset = new DatasetIris(datasetSize, inputAttributes, outputIntervals, filename);
			break;
		default:
			return nullptr;
		}

		dataset->InitializeDataset();
		dataset->NormalizeDataset();
		return dataset;
	}

Datasets::DataSets* Datasets::DatasetChooser::ReadDatasetFromFile(Datasets::DatasetType type, const std::wstring& filename)
	{
		DataSets* dataset = nullptr;
		switch (type)
		{
		case Datasets::DatasetType::Iris:
			dataset = new DatasetIris(filename);
			break;
		default:
			return nullptr;
		}

		dataset->InitializeDataset();
		dataset->NormalizeDataset();
		return dataset;
	}

